pkg/session/session.go

```diff
@@ -1641,8 +1641,12 @@ func (s *session) ExecuteInternal(ctx context.Context, sql string, args ...inter
 	if err != nil {
 		return nil, err
 	}
+	if sq, ok := stmtNode.(*ast.NonTransactionalDMLStmt); ok {
+		rs, err = HandleNonTransactionalDML(ctx, sq, s)
```

Comment:
Not sure if this is a good idea. You definitely need more tests and consideration.
Some of the questions that occurred to me:
Is `ExecuteInternal` the right place to put the entrance of `HandleNonTransactionalDML`? How about `ExecuteInternalStmt`, `Execute` and `ExecuteStmt`? 
Non-transactional DML calls `ExecuteStmt`. The entrance was put outside the `session` interface because `ExecuteStmt` is not supposed to be recursively called. Is it safe to put it in `ExecuteInternal`? Should we leave a comment to explain? Should we introduce any check to prevent possible recursion?

Reply:
I think it can be simplified into a for loop with a limit.